### ISSUE TITLE
Support upload crash report

### ISSUE THREAD

**Issue by rogerwang** [nwjs/nw.js#5533]
We need to support: 1) uploading crash dump once they are generated 2) app developers can query the crash IDs like 'chrome://crashes/'

Chromium has this feature but was disabled in NW. We need to provide a way to let app developers customize it (e.g. upload url).

**Comment by mscreenie** [comment]
Great feature. I was thinking of this just a day ago. Will make debugging much easier.

**Comment by lunemec** [comment]
Hello, I wanted to use NW feature for sending minidumps to a breakpad-server, however there is some problem with it.

Steps to replicate:
1. Start https://github.com/acrisci/simple-breakpad-server locally
2. Set crash_report_url (http://docs.nwjs.io/en/latest/References/Manifest%20Format/#crash_report_url) to http://localhost:1127/crashreports
3. Crash the nw.js app.
4. Navigate to http://localhost:1127 -> where Product/Version are not filled and minidump file cannot be downloaded because it doesn't exist.

When you upload *any* file with curl:
```
curl -F "type=multipart/form-data" -F "prod=myapp" -F "ver=3.5.1" -F "upload_file_minidump=@somefile" http://localhost:1127/crashreports
```

and goto `http://localhost:1127`, the Product/Version are correctly displayed, and the file can be downloaded, which means it was successfully uploaded.

I believe nw.js sends incorrectly formatted form/multipart data, because I was seeing strange parse errors when I tried to write Golang server for storing minidumps.

Thank you!

**Comment by lunemec** [comment]
cc @rogerwang have you tried this feature? send a crash dump directly from NW.js to some breakpad server? I know there is Python test server here, but since it doesn't work with `simple-breakpad-server` or my Go server, I guess there might be a bug in the python implementation ...

**Comment by rogerwang** [comment]
@lunemec yes. This feature is working and it's in our CI test: https://github.com/nwjs/nw.js/tree/nw24/test/sanity/crash-dump-report

**Comment by janRucka** [comment]
@rogerwang: Did you test sending dumps from Windows as well?

**Comment by rogerwang** [comment]
Sure, the CI tests run on all platforms.
